artix7/CMakeLists.txt

```diff
@@ -1,22 +1,53 @@
 include(make/project_xray.cmake)
 
 add_subdirectory(primitives)
+
+# Convert prjxray database into VPR channels, pin definitions for tile types,
+# and create direct connection list for arch.xml.
+project_xray_prepare_database(
+  PART artix7
+  )
+
+
 add_subdirectory(tiles)
 add_subdirectory(blocks)
 
+
 define_arch(
   ARCH artix7
-  YOSYS_SCRIPT "synth_xilinx -vpr"
+  YOSYS_SCRIPT "synth_xilinx -vpr -flatten"
```

Comment:
Can you add a comment about why `-flatten` is needed?

Reply:
Because if it isn't flattened, the eblif has multiple modules, which confuses things.  ice40 currently uses -flatten too.